core/primitives/src/test_utils.rs

```diff
@@ -261,19 +262,22 @@ impl BlockHeader {
             BlockHeader::BlockHeaderV1(_) | BlockHeader::BlockHeaderV2(_) => {
                 panic!("old header should not appear in tests")
             }
-            BlockHeader::BlockHeaderV3(header) => header,
+            BlockHeader::BlockHeaderV3(header) => Arc::make_mut(header),
```

Comment:
This looks wrong.

```rust
let mut header: BlockHeader = ...;
modify_header_v3(header.get_mut());  // ← essentially a nop
// At this point header should be modified but it won’t.
```

You will need to update pointer in self.

Applies to other _mut and set methods in this PR.